roleplay.py

```diff
@@ -75,32 +171,53 @@ class Roleplay(BasePlugin):
+                 "-r/--remove: Removes speficied roles from the list.\n"
+                 "Calling it without any arguments prints the list.",
+             syntax="[-a/--add (role mentions/ids/names)] [-r/--remove (role mentions/ids/names)]",
              perms={"manage_messages"},
              category="role_play")
     async def _racerole(self, msg):
         gid = str(msg.guild.id)
         self._initialize(gid)
-        args = shlex.split(msg.content)
-        if len(args) < 2:
+
+        parser = RSArgumentParser()
+        parser.add_argument("command")
+        parser.add_argument("-a", "--add", default=[], nargs='+')
+        parser.add_argument("-r", "--remove", default=[], nargs='+')
+
+        args = parser.parse_args(shlex.split(msg.content))
+
+        if not (args['add'] or args['remove']):
             await split_output(msg, "**ANALYSIS: Currently approved race roles:**",
                                [x.name for x in msg.guild.roles if x.id in self.plugin_config[gid]["race_roles"]])
         else:
-            if args[1].lower() == "add":
-                for arg in args[1:]:
-                    t_role = find_role(msg.guild, arg)
-                    if t_role and t_role.id not in self.plugin_config[gid]["race_roles"]:
-                        self.plugin_config[gid]["race_roles"].append(t_role.id)
-                await respond(msg, "**AFFIRMATIVE. Roles added to race list.**")
-            elif args[1].lower() == "remove":
-                for arg in args[1:]:
-                    t_role = find_role(msg.guild, arg)
-                    if t_role and t_role.id not in self.plugin_config[gid]["race_roles"]:
-                        self.plugin_config[gid]["race_roles"].remove(t_role.id)
-                await respond(msg, "**AFFIRMATIVE. Roles removed from race list.**")
+            args['add'] = [r for r in [find_role(msg.guild, r) for r in args['add']] if r]
+            args['remove'] = [r for r in [find_role(msg.guild, r) for r in args['remove']]if r]
+
+            # for nice output
+            l_add = []
+            l_rem = []
+
+            for role in args['add']:
+                if role.id not in self.plugin_config[gid]["race_roles"]:
+                    l_add.append(role.name)
+                    self.plugin_config[gid]["race_roles"].append(role.id)
+            for role in args['remove']:
+                if role.id in self.plugin_config[gid]["race_roles"]:
+                    l_rem.append(role.name)
+                    self.plugin_config[gid]["race_roles"].remove(role.id)
+
+            if l_add or l_rem:
+                _s = "**AFFIRMATIVE. ANALYSIS:**\n```diff\n"
+                if l_add:
+                    _s += "Added roles:\n+ " + "\n+ ".join(l_add) + "\n"
+                if l_rem:
+                    _s += "Removed roles:\n- " + "\n- ".join(l_rem) + "\n"
+                _s += "```"
+                await respond(msg, _s)
```

Comment:
More verbose variable strings would be preferred here; something like `result_str`, `role_add_list`, `role_remove_list` for example.